file-attachments.mdx

```diff
@@ -0,0 +1,184 @@
+---
+ title: 'File Attachments: Databases can now store files and images'
+ description: 'Extending a database record to include a new file column type with a global CDN, common security boundaries and image transformations.'
+ image:
+   src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/fa_blog_illustration.jpeg
+   alt: 'File Attachments: Databases can now store files and images'
+ author: Sorin Toma
+ date: 08-30-2023
+ published: true
+ slug: file-attachments
+ ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/fa_blog_illustration.jpeg
+---
+
+Today, as part of [our launch week](https://xata.io/blog/launch-week-august-2023), we’re beyond excited to announce a feature that we've wanted to add ever since we started Xata: File Attachments. Think of it as having a new database column type where you can store files of any size, and behind the scenes they are stored in AWS S3 and cached through a global CDN. With this approach, files simply become part of a database record. For example, they respect the same security boundary -- if you delete a record, the files also get deleted. Image file types also get some extra functionality allowing you to request them at any size and style with built-in transformations.
+
+Whether you’re building a full scale e-commerce application, a personal blog or evolving your product to be AI-native — file attachments unlock a number of new use cases while providing an opportunity to simplify your architecture and reduce the number of services you rely on.
```

Comment:
This is a bit repetitive with a line in the next section "Think about a blog which uses both posts metadata and images, a product catalogue or a document library."

And it's a bit market-y, I worry a bit about it's impact on the HN audience.

Reply:
Removed the marketing-y piece and left the problem we're trying to solve

> Today, as part of [our launch week](https://xata.io/blog/launch-week-august-2023), we’re beyond excited to announce a feature that we've wanted to add ever since we started Xata: File Attachments. Think of it as having a new database column type where you can store files of any size, and behind the scenes they are stored in AWS S3 and cached through a global CDN. Files simply become a part of a database record. For example, they respect the same security boundary -- if you can access a record, you can also access its attached files. Image file types also get some extra functionality allowing you to request them at any size and style with built-in transformations. With this release, we aim to simplify your application architecture and reduce the number of services you need to manage.
> 
> In this blog we'll dive into the capabilities released today and architecture behind our implementation. We hope you enjoy the read as much as we did building it 😃